algorithms/depth_first.py

```diff
@@ -0,0 +1,70 @@
+class BinaryTree():
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+    def child(self, data):
```

Comment:
```suggestion
    def insert(self, data):
```
Name functions by their purpose